packages/caliper-fabric/lib/identity-management/IdentityManager.js

```diff
@@ -184,7 +184,14 @@ class IdentityManager {
      * @private
      */
     async _extractIdentitiesFromWallet(mspId, wallet) {
-        // TODO: To be implemented
+        const walletFacade = await this.walletFacadeFactory.create(wallet.path);
+        const allIDNames = await walletFacade.getAllIdentityNames();
+        for (const identityNames in allIDNames){
+            const identity = await walletFacade.export(identityNames);
+            if (identity){
```

Comment:
we can remove this check as we should expect something to be returned as the wallet has given the list of all available identity names